src/bsec/inc/bsec_datatypes.h

```diff
@@ -0,0 +1,489 @@
+ * application work. The Software and Information is subject to the following terms and conditions:
+ *
+ * The Software is specifically designed for the exclusive use for Bosch Sensortec products by
+ * personnel who have special experience and training. Do not use this Software if you do not have the
+ * proper experience or training.
+ *
+ * This Software package is provided `` as is `` and without any expressed or implied warranties,
+ * including without limitation, the implied warranties of merchantability and fitness for a particular
+ * purpose.
+ *
+ * Bosch Sensortec and their representatives and agents deny any liability for the functional impairment
+ * of this Software in terms of fitness, performance and safety. Bosch Sensortec and their
+ * representatives and agents shall not be liable for any direct or indirect damages or injury, except as
+ * otherwise stipulated in mandatory applicable law.
+ *
+ * The Information provided is believed to be accurate and reliable. Bosch Sensortec assumes no
+ * responsibility for the consequences of use of such Information nor for any infringement of patents or
+ * other rights of third parties which may result from its use. No license is granted by implication or
+ * otherwise under any patent or patent rights of Bosch. Specifications mentioned in the Information are
+ * subject to change without notice.
+ *
+ * It is not allowed to deliver the source code of the Software to any third party without permission of
+ * Bosch Sensortec.
+ *
+ */
+
+ /**
+ * @file        bsec_datatypes.h
+ *
+ * @brief
+ * Contains the data types used by BSEC
+ *
+ */
+
+#ifndef __BSEC_DATATYPES_H__
+#define __BSEC_DATATYPES_H__
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/*!
+ * @addtogroup bsec_interface BSEC C Interface
+ * @{*/
+
+#ifdef __KERNEL__
+#include <linux/types.h>
+#endif
+#include <stdint.h>
+#include <stddef.h>
+
+#define BSEC_MAX_WORKBUFFER_SIZE     (2048)    /*!< Maximum size (in bytes) of the work buffer */
+#define BSEC_MAX_PHYSICAL_SENSOR     (8)         /*!< Number of physical sensors that need allocated space before calling bsec_update_subscription() */
+#define BSEC_MAX_PROPERTY_BLOB_SIZE  (454)     /*!< Maximum size (in bytes) of the data blobs returned by bsec_get_configuration() */
+#define BSEC_MAX_STATE_BLOB_SIZE     (139)        /*!< Maximum size (in bytes) of the data blobs returned by bsec_get_state()*/
+#define BSEC_SAMPLE_RATE_DISABLED    (65535.0f)      /*!< Sample rate of a disabled sensor */
+#define BSEC_SAMPLE_RATE_ULP         (0.0033333f)           /*!< Sample rate in case of Ultra Low Power Mode */
+#define BSEC_SAMPLE_RATE_CONTINUOUS  (1.0f)    /*!< Sample rate in case of Continuous Mode */ 
+#define BSEC_SAMPLE_RATE_LP          (0.33333f)            /*!< Sample rate in case of Low Power Mode */
+#define BSEC_SAMPLE_RATE_ULP_MEASUREMENT_ON_DEMAND         (0.0f)            /*!< Input value used to trigger an extra measurment (ULP plus) */        

```

Comment:
```suggestion
#define BSEC_SAMPLE_RATE_ULP_MEASUREMENT_ON_DEMAND         (0.0f)            /*!< Input value used to trigger an extra measurement (ULP plus) */        

```

Reply:
An alternative would be to configure the spellcheck to skip `./src/bsec` by adding that path to the `skip` field of a `.codespellrc` [configuration file](https://github.com/codespell-project/codespell#using-a-config-file).

Typically a `.codespellrc` file is added to the repo at the time the workflow is installed, but I see it is not currently present in this repo. The standardized base file used in Arduino repositories is available here:

https://github.com/arduino/tooling-project-assets/blob/main/workflow-templates/assets/spell-check/.codespellrc

Reply:
Good idea @per1234, I added a .codespellrc to skip parsing that directory since I'd like to avoid touching/maintaining it. We're going to remove it when the upstream lib is fixed hopefully...